pkg/operator/etcdendpointscontroller/etcdendpointscontroller.go

```diff
@@ -124,3 +166,57 @@ func configMapAsset() *corev1.ConfigMap {
 		},
 	}
 }
+
+// isSafeToRemoveBootstrapIP returns true if it's safe to remove the bootstrap
+// member IP from the endpoints list, otherwise false. If the determination
+// can't be made, returns false and an error.
+func (c *EtcdEndpointsController) isSafeToRemoveBootstrapIP() (bool, error) {
+	// See if the etcd client knows about the bootstrap member.
+	members, err := c.etcdClient.MemberList()
+	if err != nil {
+		return false, fmt.Errorf("couldn't list etcd members: %w", err)
+	}
+	bootstrapFound := false
+	for _, member := range members {
+		if member.Name == "etcd-bootstrap" {
+			bootstrapFound = true
+			break
+		}
+	}
+	return !bootstrapFound && len(members) >= 3, nil
+}
+
+// tryRemoveBootstrapIP will remove the bootstrap IP annotation from the endpoints
+// configmap. If bootstrapping is detected to still be in progress, the function
+// does nothing and returns no error to facilitate quiet retries.
+func (c *EtcdEndpointsController) tryRemoveBootstrapIP(ctx context.Context, recorder events.Recorder) error {
+	existing, err := c.configmapLister.ConfigMaps(operatorclient.TargetNamespace).Get("etcd-endpoints")
+	if err != nil {
+		return fmt.Errorf("couldn't get required configmap %s/%s: %w", operatorclient.TargetNamespace, "etcd-endpoints", err)
+	}
+
+	// Only proceed if there's actually something to do.
+	if _, existingHasBootstrapIP := existing.Annotations[etcdcli.BootstrapIPAnnotationKey]; !existingHasBootstrapIP {
+		return nil
+	}
+
+	// See if the etcd client knows about the bootstrap member.
+	isSafe, err := c.isSafeToRemoveBootstrapIP()
+	if err != nil {
+		return fmt.Errorf("failed to determine whether it's safe to remove the boostrap IP annotation: %w", err)
+	}
+	// If it's not yet safe to remove the annotation, no-op and we'll try
+	// again during another sync.
+	if !isSafe {
+		return nil
+	}
+
+	// Bootstrap appears to be complete, so remove the bootstrap IP annotation.
+	updated := existing.DeepCopy()
+	delete(updated.Annotations, etcdcli.BootstrapIPAnnotationKey)
```

Comment:
merging of metadata will remove annotations if they are prefixed with a `-` I think.  have a read.